erts/emulator/beam/erl_bits.c

```diff
@@ -1412,6 +1406,19 @@ erts_bs_append(Process* c_p, Eterm* reg, Uint live, Eterm build_size_term,
 	    }
 	}
 
+	if (build_size_in_bits == 0) {
+	    goto return_bin;
+	}
+
+	/*
+	 * Allocate heap space.
+	 */
```

Comment:
I just moved this down so we don't allocate more than we are supposed to in case nothing is being appended.